lang/cs/Org.Apache.REEF.Common/Telemetry/MetricBase.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Threading;
+using Newtonsoft.Json;
+
+namespace Org.Apache.REEF.Common.Telemetry
+{
+    /// <summary>
+    /// Base implementation of a metric with untyped value.
+    /// </summary>
+    public class MetricBase : IMetric
+    {
+        protected ITracker _tracker;
+
+        private object _value;
+
+        protected object _metricLock = new object();
+
+        public string Name
+        {
+            get; internal set;
```

Comment:
these should be private set right? Ideally you want only the constructor to set these and not allow anyone to modify.

Reply:
This is used set properties when IEvaluatorMetrics calls CreateAndRegisterMetric<T> where the metric's parameterless constructor is called.

Reply:
OK - see comment above in the CreateAndRegisterMetric function. I think if you change that to calling the constructor instead of setting the properties it will work.